python/lsst/rubintv/production/pipelineRunning.py

```diff
@@ -607,6 +626,59 @@ def postProcessVisitSummary(self, quantum: Quantum) -> None:
         except Exception:
             self.log.exception("Failed to populate visit row in ConsDB")
 
+    def postProcessCalcZernikes(self, quantum: Quantum) -> None:
+        """Post-process the Zernike table to send results to ConsDB."""
+        # protect import to stop the whole package depending on ts_wep. If this
+        # becomes a problem we could copy the functions or just accept that RA
+        # needs T&S software.
+        from lsst.ts.wep.utils.zernikeUtils import makeDense
+
+        try:
+            dRef = quantum.outputs["zernikes"][0]
+            zkTable = self.cachingButler.get(dRef)
+        except Exception:
+            self.log.warning(
+                f"Failed to post-process *failed* quantum {quantum}. This is not unexpected"
+                " but still merits a warning due to the failing quantum."
+            )
+            return
+
+        # *ideally* this would be pulled from maxconfig.nollIndices() but
+        # that's not possible here, but a) they never run above 28, and b)
+        # ConsDB only goes out that far, so we'd truncate there anyway, so we
+        # just hardcode it here.
+        MAX_NOLL_INDEX = 28
+
+        # Get the physical rotation from the EFD. Ideally this would be pulled
+        # from the ConsDB, but that's calculated elsewhere in RA, and although
+        # that process is much quicker, using it here is introducing an
+        # unnecessary race condition, so it's better to recalculate it here.
+        visitRecord = dRef.dataId.records["visit"]
+        assert visitRecord is not None, "visitRecord is None, this shouldn't be possible"
+        data = getEfdData(self.efdClient, "lsst.sal.MTRotator.rotation", expRecord=visitRecord)
+        physicalRotation = np.nanmean(data["actualPosition"])
+
+        detector = dRef.dataId.records["detector"]  # not a detector object, but a detector dimension
+        assert detector is not None, "detector is None, this shouldn't be possible"
+        detectorId: int = detector.id  # hence .id rather than .getId()
+
+        zkTable = zkTable[zkTable["label"] == "average"]
+        zkColsHere = [col for col in zkTable.colnames if col.startswith("Z")]
+        nollIndicesHere = np.asarray([int(col.removeprefix("Z")) for col in zkColsHere], dtype=int)
+        # Grab Zernike values, convert to dense array, save
+        zkSparse = zkTable[zkColsHere].to_pandas().values[0]
+        zkDense = makeDense(zkSparse, nollIndicesHere, MAX_NOLL_INDEX)
+        rotationMatrix = zernikeRotMatrix(MAX_NOLL_INDEX, -np.deg2rad(physicalRotation))
+        zernikeValues: np.ndarray = zkDense / 1e3 @ rotationMatrix[4:, 4:]
```

Comment:
Why are the lower values dropped? No need to address if this should be obvious to someone working with this code, but it's not apparent to a newbie.

Reply:
Because we are only really keeping track of z4 and above. You could add a comment there about this.

Reply:
Thanks - will add a comment. ConsDB only has slots for z4 upwards anyway, so that's justification enough 🙂